cltk/lemmatize/latin/unsupervised.py

```diff
@@ -0,0 +1,382 @@
+        MD5 hash of the file.
+
+
+    """
+    def __init__(self, path, mode='r', buffer=True, validate=False):
+        self.path = path
+        self.mode = mode
+        self.buffer = buffer
+        self.fname = os.path.basename(path)
+
+        if buffer:
+            self.file = open(path, 'r')
+        else:
+            self.file = []
+            with open(path, 'r') as f:
+                for line in f.readlines():
+                    self.file.append(line)
+
+        self.__hash = None
+        self.__len = None
+
+        if validate:
+            self.validate()
+
+    def __getitem__(self, index):
+        if index < 0 or index >= len(self):
+            raise IndexError()
+
+        if self.buffer:
+            self.file.seek(0)
+            for _ in range(index + 1):
+                line = self.file.readline()
+            return line
+        return self.file[index]
+
+    def __len__(self):
+        if self.__len is None:
+            self.__len = sum([1 for _ in self.readlines()])
+        return self.__len
+
+    @property
+    def hash(self):
+        """The MD5 hash of the .tess file"""
+        if self.__hash is None:
+            hashinator = hashlib.md5()
+            for line in self.readlines():
+                hashinator.update(line.encode('utf-8'))
+            self.__hash = hashinator.hexdigest()
+        return self.__hash
+
+    def readlines(self):
+        """Iterate over the lines of the .tess file in order.
+
+        Yields
+        ------
+        line : str
+            One line of the .tess file.
+        """
+        if self.buffer:
+            self.file.seek(0)
+            for line in self.file.readlines():
+                yield line
+        else:
+            for line in self.file:
+                yield line
+
+    def read_tokens(self, include_tag=False):
+        """Iterate over the tokens of a .tess file in order.
+
+        Parameters
+        ----------
+        include_tag : bool
+            If True, include the starting tag with each line of the .tess file.
+            Otherwise, only return the raw tokens.
+
+        Yields
+        ------
+        token : str
+            One token of the .tess file.
+        """
+        for line in self.readlines():
+            start = line.find('>') + 1 if not include_tag else 0
+            line = line[start:].strip(string.whitespace)
+            tokens = line.split()
+            for token in tokens:
+                yield token
+
+    def validate(self):
+        """Determine if this file is a valid .tess file.
+
+        Raises
+        ------
+        MalformedTessFileError
+            If a tag in the file does not contain the proper information.
+        """
+        name, ext = os.path.splitext(self.fname)
+
+        # Ensure that the file has the .tess extension
+        if ext != '.tess':
+            msg = 'Bad filename {}. tess files must end in .tess'.format(self.fname)
+            warnings.warn(msg, warning.UserWarning)
```

Comment:
I like your use of warnings, this is excellent and we will do this more elsewhere in the CLTK codebase.

However, I don't see an import warnings, so please add that.